librz/bin/p/bin_pe64.c

```diff
@@ -430,20 +491,22 @@ static RzList *trycatch(RzBinFile *bf) {
 		}
 
 		ut32 handler;
-		if (!rz_io_read_at_mapped(io, exceptionDataOff, (ut8 *)&handler, sizeof(handler))) {
+		if (!rz_buf_read_at(bin->b, exceptionDataOff, (ut8 *)&handler, sizeof(handler))) {
 			continue;
 		}
 		exceptionDataOff += sizeof(ut32);
 
 		PE64_SCOPE_TABLE tbl;
-		if (!rz_io_read_at_mapped(io, exceptionDataOff, (ut8 *)&tbl, sizeof(tbl))) {
+		if (!rz_buf_read_at(bin->b, exceptionDataOff, (ut8 *)&tbl, sizeof(tbl))) {
 			continue;
 		}
 
+		const ut64 last_scope_addr = RZ_MIN(rz_buf_size(bin->b), unwind_data_section->paddr + unwind_data_section->size) - sizeof(PE64_SCOPE_RECORD);
 		PE64_SCOPE_RECORD scope;
 		ut64 scopeRecOff = exceptionDataOff + sizeof(tbl);
-		for (i = 0; i < tbl.Count; i++) {
-			if (!rz_io_read_at_mapped(io, scopeRecOff, (ut8 *)&scope, sizeof(PE64_SCOPE_RECORD))) {
+		int i;
+		for (i = 0; i < tbl.Count && scopeRecOff <= last_scope_addr; i++, scopeRecOff += sizeof(PE64_SCOPE_RECORD)) {
+			if (!rz_buf_read_at(bin->b, scopeRecOff, (ut8 *)&scope, sizeof(PE64_SCOPE_RECORD))) {
```

Comment:
Such reads are endianness-dependent, please use the endian-independent functions. Otherwise, we increase the things we need to refactor to make PE parsing pass on the big-endian platforms, like System-Z: https://github.com/rizinorg/rizin/issues/297